Add new field for IP netmask

sebix opened the pull request:
It's not well tested; therefore, it's a draft. Creating with a divergent netmask already works.

sebix commented:
I'm not sure yet about the wording. "IP Network" or "IP Netmask"?

What values should it accept? I now implemented the CIDR notation because it is the simplest variant. We could also accept a leading `/`. Should we also accept netmask notations (`255.255.255.0`)?

Successfully tested so far:
- Add single device finding
- Edit Device finding
- Std Import (NMAP)
- Import (CSV)
- Create Device based on DeviceFinding
- View Device Finding
- Table view of device findings
- Split Selected
- Export (CSV)

sebix commented:
Validation now removes a leading `/` and makes sure the netmask is >= 0 and <= 32.

Tested for: Mapping import (NMAP), Std Import (CSV), Add Device Finding and Edit Device Finding

sebix commented:
Thinking about the naming again: I think `ip_network` is better, in line with with wording `ip_address` and already implies the CIDR notation.

Crubumble commented on plugins/d3c/models.py:
```diff
@@ -476,6 +476,13 @@ class DeviceFinding(NetBoxModel):
         verbose_name='IP Address'
     )
+    ip_netmask = models.CharField(
+        max_length=1000,
```
why not a more reasonable value?

sebix replied:
Because most other fields like IP addresse and mac address also have a maximum length of 1000. So, I used the existing convention.

sebix commented:
Because of these:

> It's not well tested; therefore, it's a draft.

> Thinking about the naming again: I think `ip_network` is better, in line with with wording `ip_address` and already implies the CIDR notation.

sebix commented:
And also this open question:

> What values should it accept? I now implemented the CIDR notation because it is the simplest variant. We could also accept a leading /. Should we also accept netmask notations (255.255.255.0)?

Crubumble commented:
Since NetBox displays the CIDR notation. CDIR notation is sufficient for now. The nomenclature has to be decided. However it also can easily be changed later. The main reason for the draft status is that some aspects are missing, which I want to add myself in the pull request

sebix commented:
TODOs here:

- [ ] If possible display the IP address and the netmask in one column as in `/ipam/ip-addresses/`
- [ ] extract the IP netmask from the IP address if given in the original data

sebix commented:
>  If possible display the IP address and the netmask in one column as in /ipam/ip-addresses/

The IPAM module has one field for IP address + netmask, so this was not a reference for the code.
Implemented for all tables, based on https://django-tables2.readthedocs.io/en/latest/pages/custom-data.html#subclassing-column

> extract the IP netmask from the IP address if given in the original data

Implemented as well, tested for add, edit, import and stdimport
